File: src/main/java/com/github/doomsdayrs/jikan4java/types/main/schedule/Day.java
package com.github.doomsdayrs.jikan4java.types.main.schedule;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.ArrayList;

/*
 * This file is part of Jikan4java.
 *
 * Jikan4java is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Jikan4java is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Jikan4java.  If not, see <https://www.gnu.org/licenses/>.
 * ====================================================================*/

/**
 * Jikan4java
 * 01 / November / 2018
 *
 * @author github.com/doomsdayrs
 */
public class Day {
    public String request_hash;
    public boolean request_cached;
    public int request_cache_expiry;
    public ArrayList<SubAnime> subAnimes;

    public Day(String request_hash, boolean request_cached, int request_cache_expiry, ArrayList<SubAnime> subAnimes) {
        this.request_hash = request_hash;
        this.request_cached = request_cached;
        this.request_cache_expiry = request_cache_expiry;
        this.subAnimes = subAnimes;
    }

    @Override
    public String toString() {
        return "Day{" +
                "request_hash='" + request_hash + '\'' +
                ", request_cached=" + request_cached +
                ", request_cache_expiry=" + request_cache_expiry +
                ", subAnimes=" + subAnimes +
                '}';
    }
}

File: src/commonMain/kotlin/com/github/doomsdayrs/jikan4java/data/model/main/schedule/Day.kt
package com.github.doomsdayrs.jikan4java.data.model.main.schedule

import com.github.doomsdayrs.jikan4java.data.model.support.RequestHashing
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/*
 * This file is part of Jikan4java.
 *
 * Jikan4java is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Jikan4java is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Jikan4java.  If not, see <https://www.gnu.org/licenses/>.
 * ====================================================================*/
/**
 * Jikan4java
 * 01 / November / 2018
 *
 * @author github.com/doomsdayrs
 */
@Serializable
sealed class Day : RequestHashing {
	abstract val subAnimes: List<SubAnime>

	@Serializable
	data class Friday(
		@SerialName("request_hash") override val requestHash: String,
		@SerialName("request_cached") override val requestCached: Boolean,
		@SerialName("request_cache_expiry") override val requestCacheExpiry: Int,
		@SerialName("friday") override val subAnimes: List<SubAnime>
	) : Day()


	@Serializable
	data class Monday(
		@SerialName("request_hash") override val requestHash: String,
		@SerialName("request_cached") override val requestCached: Boolean,
		@SerialName("request_cache_expiry") override val requestCacheExpiry: Int,
		@SerialName("monday") override val subAnimes: List<SubAnime>
	) : Day()

	@Serializable
	data class Other(
		@SerialName("request_hash") override val requestHash: String,
		@SerialName("request_cached") override val requestCached: Boolean,
		@SerialName("request_cache_expiry") override val requestCacheExpiry: Int,
		@SerialName("other") override val subAnimes: List<SubAnime>
	) : Day()

	@Serializable
	data class Saturday(
		@SerialName("request_hash") override val requestHash: String,
		@SerialName("request_cached") override val requestCached: Boolean,
		@SerialName("request_cache_expiry") override val requestCacheExpiry: Int,
		@SerialName("saturday") override val subAnimes: List<SubAnime>
	) : Day()

	@Serializable
	data class Sunday(
		@SerialName("request_hash") override val requestHash: String,
		@SerialName("request_cached") override val requestCached: Boolean,
		@SerialName("request_cache_expiry") override val requestCacheExpiry: Int,
		@SerialName("sunday") override val subAnimes: List<SubAnime>
	) : Day()

	@Serializable
	data class Thursday(
		@SerialName("request_hash") override val requestHash: String,
		@SerialName("request_cached") override val requestCached: Boolean,
		@SerialName("request_cache_expiry") override val requestCacheExpiry: Int,
		@SerialName("thursday") override val subAnimes: List<SubAnime>
	) : Day()

	@Serializable
	data class Tuesday(
		@SerialName("request_hash") override val requestHash: String,
		@SerialName("request_cached") override val requestCached: Boolean,
		@SerialName("request_cache_expiry") override val requestCacheExpiry: Int,
		@SerialName("tuesday") override val subAnimes: List<SubAnime>
	) : Day()

	@Serializable
	data class Unknown(
		@SerialName("request_hash") override val requestHash: String,
		@SerialName("request_cached") override val requestCached: Boolean,
		@SerialName("request_cache_expiry") override val requestCacheExpiry: Int,
		@SerialName("unknown") override val subAnimes: List<SubAnime>
	) : Day()

	@Serializable
	data class Wednesday(
		@SerialName("request_hash") override val requestHash: String,
		@SerialName("request_cached") override val requestCached: Boolean,
		@SerialName("request_cache_expiry") override val requestCacheExpiry: Int,
		@SerialName("wednesday") override val subAnimes: List<SubAnime>
	) : Day()
}